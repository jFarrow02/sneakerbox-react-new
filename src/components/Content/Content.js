import React from 'react';
import './Content.scss';
import {
    Home,
    MyAccount,
    ProductThumbnails,
} from '../index';
import {Route} from 'react-router-dom';
import {navConfig} from '../../data/config';

const resolveComponentName = (name, props)=>{
    const map = {
        ProductThumbnails: props ? <ProductThumbnails {...props}/> : <ProductThumbnails/>,
        Home: props ? <Home {...props}/> : <Home/>,
        MyAccount: props ? <MyAccount {...props}/> : <MyAccount/>,
    };
    return map[name];
}

const Content = ()=>{

    return(   
        <main className='Content'>
            Content goes here
            {
                navConfig.map((conf, idx)=>{
                    const productsTest = /\/products\//;
                    if(productsTest.test(conf.url)){
                        return(
                            <Route
                                key={`route-${idx}`}
                                path={conf.url}
                            >
                                {resolveComponentName(conf.component, {category: conf.url.split('/products/')[1]})}
                            </Route>
                        )
                    }
                    else{
                        return <Route key={`route-${idx}`} path={conf.url}>{resolveComponentName(conf.url)}</Route>
                    }
                })
            }
            
        </main>
    );
}

export default Content;