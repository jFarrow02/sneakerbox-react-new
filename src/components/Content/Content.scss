@import '../../styles/variables.scss';

.Content{
    grid-column: 4 /span 9;
    grid-row: 2 /span 1;
    background-color: $offWhite;
}