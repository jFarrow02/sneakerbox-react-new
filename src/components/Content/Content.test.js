import React from 'react';
import Content from './Content';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';

it('should match a snapshot', ()=>{
    const tree = renderer.create(<BrowserRouter><Content/></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});