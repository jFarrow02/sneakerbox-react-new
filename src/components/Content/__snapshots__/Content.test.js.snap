// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<main
  className="Content"
>
  <section
    className="Home"
  >
    <div
      className="CaptionedImage"
    >
      <div
        className="CaptionedImage--caption"
        style={
          Object {
            "left": "15px",
            "position": "absolute",
            "top": "15px",
          }
        }
      >
        <div
          className="Home--caption"
        >
          <h2>
            Winter Mode
          </h2>
          <p>
            Get ready for rugged adventures with our new 
            <strong>
              Winter Collection.
            </strong>
          </p>
          <a
            href="#"
          >
            &gt;&gt;Shop Now.
          </a>
        </div>
      </div>
      <img
        alt="Main image home page; man walking along railway"
        height=""
        src="sneakers-railtracks.jpg"
        title=""
        width=""
      />
    </div>
  </section>
</main>
`;
